### DISCUSSION TITLE
Switching from IMv6 syntax to IMv7 results in worse output

### DISCUSSION THREAD

**Discussion by sbecuwe** [ImageMagick/ImageMagick#7631] [Help] [answered]
### ImageMagick version

7.1.1-38

### Operating system

MacOS

### Operating system, version and so on

14.7

### Description

Switching to the new syntax where the input file follows immediately the program, results in a different (worse) output compared to the old syntax. Is this is a bug or am I overlooking something? 
testold.png looks the way I expect it, testnew.png doesn't. 

### Steps to Reproduce

```
magick testpdf.pdf -density 300 -colorspace sRGB -background white -alpha remove -resize x1024 testnew.png
convert -density 300 -colorspace sRGB -background white -alpha remove -resize x1024 testpdf.pdf testold.png
```

### Images

[testpdf.pdf](https://github.com/user-attachments/files/17040414/testpdf.pdf)
![testnew](https://github.com/user-attachments/assets/f39e419c-c2c1-4d42-bed2-0b7faadb9137)
![testold](https://github.com/user-attachments/assets/634a778b-7950-4cc4-99a5-9b7f1e4ee11e)

**Comment by dlemstra** [comment, marked as answer]
With `magick` the parameters are read in the order that you specify them. I think that moving `-density` before `testpdf.pdf` will resolve your issue.
